src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -143,15 +142,18 @@ class NotebookPreviewFrame extends Component {
   }
 
   render() {
-    const { notebookName, workspace, queryParams }  = this.props
+    const { loadedChildren } = this.props
     const { preview, busy } = this.state
     return h(Fragment, [
-      !!queryParams['read-only'] && h(ReadOnlyMessage, { notebookName, workspace }),
```

Comment:
Moved so it appears if you're read-only for any reason

Reply:
When you moved the `ReadOnlyMessage` to `NotebookPreview`, I expected it to show up whenever a notebook is being previewed, like while a cluster is loading. How did you get it to show up only when a writer is viewing read-only?